examples/acquiring_data/goes_xrs_example.py

```diff
@@ -67,6 +67,14 @@
 goes_15 = ts.TimeSeries(file_goes15)
 goes_15.peek()
 
+#############################################################
+# The resulting `~sunpy.timeseries.TimeSeries` can be filtered by GOES quality flags. For more information
+# refer to the `GOES Data Guide <https://satdat.ngdc.noaa.gov/sem/goes/data/science/xrs/GOES_13-15_XRS_Science-Quality_Data_Readme.pdf>`__.
+
+df = goes_15.to_dataframe()
+df = df[(df["xrsa_quality"] == 0) & (df["xrsb_quality"] == 0)]
+goes_15.data = df
```

Comment:
```suggestion
goes_15 = ts.TimeSeries(df, goes_15.meta, goes_15.units)
```
This is the best way to do this - if we change the internals of how `TimeSeries` works in the future this will avoid relying `.data` being a dataframe.